web-client/src/modules/request/index.ts

```diff
@@ -1,10 +1,92 @@
+import {
+  ContainerOutlined,
+  FileAddOutlined,
+  FileDoneOutlined,
+  FileProtectOutlined,
+  FileSearchOutlined,
+  FileSyncOutlined,
+} from '@ant-design/icons';
+import { uniqueId } from 'lodash';
+import React from 'react';
+import Location from 'react-app-location';
 import { Module } from 'src/types/module';
 
+import { MenuItem } from '../../types/menu-item';
 import ContentPage from './pages/ContentPage';
+import { NewRequestLocation } from './pages/routes/NewRequestRoute/constants';
+
+// TODO (remove link id): ask help to find a better way to keep track of ids
+export const menuItems: Array<MenuItem> = [
+  {
+    id: uniqueId(),
+    icon: React.createElement(FileAddOutlined),
+    title: 'Create a new Request',
+    location: NewRequestLocation,
+  },
+  {
+    id: uniqueId(),
+    title: 'Find Requests',
+    children: [
+      {
+        id: uniqueId(),
+        icon: React.createElement(FileSearchOutlined),
+        title: 'Open',
+        location: new Location('/requests/open'),
+      },
+      {
+        id: uniqueId(),
+        icon: React.createElement(FileDoneOutlined),
+        title: 'Accepted',
+        location: new Location('/requests/accepted'),
+      },
+      {
+        id: uniqueId(),
+        icon: React.createElement(FileSyncOutlined),
+        title: 'Ongoing',
+        location: new Location('/requests/ongoing'),
+      },
+      {
+        id: uniqueId(),
+        icon: React.createElement(FileProtectOutlined),
+        title: 'Finished',
+        location: new Location('/requests/finished'),
+      },
+      {
+        id: uniqueId(),
+        icon: React.createElement(ContainerOutlined),
+        title: 'Archived',
+        children: [
+          {
+            id: uniqueId(),
+            title: 'Rejected',
+            location: new Location('/requests/archived?open'),
+          },
+          {
+            id: uniqueId(),
+            title: 'Canceled',
+            location: new Location('/requests/archived?canceled'),
+          },
+          {
+            id: uniqueId(),
+            title: 'Closed',
+            location: new Location('/requests/archived?closed'),
+          },
+          {
+            id: uniqueId(),
+            title: 'Completed',
+            location: new Location('/requests/archived?completed'),
+          },
+        ],
+      },
+    ],
+  },
+];
 
 const requestModule: Module = {
   path: '/requests',
   component: ContentPage,
   protected: true,
+  layout: 'dashboard',
+  menuItems,
```

Comment:
As we discussed with @ashwinkjoseph this is not required at all because most of our pages will use the same layout except login steps. But I couldn't figure out how to merge DashboardRoute and ProtectedRoute at this moment. That's why I used this for just experimentation. Maybe the DashboardRoute is unnecessary, then we can just use only the ProtectedRoute.

Reply:
this is fine and I feel is required. so let it be